models/trimmer_decorator.rb

```diff
@@ -0,0 +1,9 @@
+require_relative 'base_decorator'
+
+class TrimmerDecorator
+  def correct_name
+    return super.slice(0..9) unless super.length <= 10
+
+    super
+  end
```

Comment:
- [ ] the method should return `@nameable.correct_name[0..9]` or using your method `@nameable.correct_name.slice(0..9)`